src/lib/FieldType/Validator/TargetContentValidator.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Core\FieldType\Validator;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\FieldType\ValidationError;
+
+/**
+ * Validator for checking existence of content and its content type.
+ */
+final class TargetContentValidator implements TargetContentValidatorInterface
+{
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    private $contentTypeService;
+
+    public function __construct(
+        ContentService $contentService,
+        ContentTypeService $contentTypeService
+    ) {
+        $this->contentService = $contentService;
+        $this->contentTypeService = $contentTypeService;
+    }
+
+    /**
+     * @param mixed $value
```

Comment:
Why can't you use strict `int` type here? I think you will either expect string or integer here, you can int-cast strings upon method calls.